modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFolderItemLocalServiceImpl.java

```diff
@@ -19,4 +36,237 @@
+		}
+	}
+
+	@Override
+	public void deleteObjectFolderItemsByObjectFolderId(long objectFolderId) {
+		List<ObjectFolderItem> objectFolderItems =
+			objectFolderItemPersistence.findByObjectFolderId(objectFolderId);
+
+		for (ObjectFolderItem objectFolderItem : objectFolderItems) {
+			objectFolderItemPersistence.remove(objectFolderItem);
+		}
+	}
+
+	@Override
+	public List<ObjectFolderItem> fetchObjectFolderItemsByObjectDefinitionId(
+		long objectDefinitionId) {
+
+		return objectFolderItemPersistence.findByObjectDefinitionId(
+			objectDefinitionId);
+	}
+
+	@Override
+	public List<ObjectFolderItem> fetchObjectFolderItemsByObjectFolderId(
+		long objectFolderId) {
+
+		return objectFolderItemPersistence.findByObjectFolderId(objectFolderId);
+	}
+
+	@Override
+	public ObjectFolderItem getObjectFolderItem(
+			long objectDefinitionId, long objectFolderId)
+		throws PortalException {
+
+		return objectFolderItemPersistence.findByPrimaryKey(
+			new ObjectFolderItemPK(objectDefinitionId, objectFolderId));
+	}
+
+	@Override
+	public ObjectFolderItem updateObjectFolderItem(
+			long objectDefinitionId, long objectFolderId, int positionX,
+			int positionY)
+		throws PortalException {
+
+		ObjectFolderItem objectFolderItem =
+			objectFolderItemPersistence.findByPrimaryKey(
+				new ObjectFolderItemPK(objectDefinitionId, objectFolderId));
+
+		objectFolderItem.setPositionX(positionX);
+		objectFolderItem.setPositionY(positionY);
+
+		return objectFolderItemPersistence.update(objectFolderItem);
+	}
+
+	@Override
+	public void updateObjectFolderItem(
+			long objectDefinitionId, long newObjectFolderId,
+			long oldObjectFolderId)
+		throws PortalException {
+
+		if (newObjectFolderId == oldObjectFolderId) {
+			return;
+		}
+
+		ObjectDefinition objectDefinition =
+			_objectDefinitionPersistence.findByPrimaryKey(objectDefinitionId);
+
+		_updateObjectFolderItem(
+			objectDefinitionId, newObjectFolderId, oldObjectFolderId,
+			objectDefinition.getUserId());
+
+		for (ObjectDefinition relatedObjectDefinition :
+				_getRelatedObjectDefinitions(objectDefinition)) {
+
+			_updateObjectFolderItem(
+				relatedObjectDefinition.getObjectDefinitionId(),
+				newObjectFolderId, oldObjectFolderId,
+				objectDefinition.getUserId());
+		}
+	}
+
+	private List<ObjectDefinition> _getRelatedObjectDefinitions(
+		ObjectDefinition objectDefinition) {
+
+		List<ObjectDefinition> relatedObjectDefinitions = new ArrayList<>();
+
+		for (ObjectRelationship objectRelationship :
+				ListUtil.concat(
+					_objectRelationshipPersistence.findByODI1_R(
+						objectDefinition.getObjectDefinitionId(), false),
+					_objectRelationshipPersistence.findByODI2_R(
+						objectDefinition.getObjectDefinitionId(), false))) {
+
+			if (objectRelationship.isSelf()) {
+				continue;
+			}
+
+			relatedObjectDefinitions.add(
+				ObjectRelationshipUtil.getRelatedObjectDefinition(
+					objectDefinition, objectRelationship));
+		}
```

Comment:
Please, see this feedback from Brian from a previous review: https://github.com/brianchandotcom/liferay-portal/commit/2a91798016e3edcb4265da3039ad3157ba7ded7c

Reply:
I saw this comment and I tried to implement this days ago. But in my option is different things. This method is responsibly to return a List of definitions. In Brian's commit is a void method... What do you think?